src/WormholeL2ERC20.sol

```diff
@@ -80,8 +80,7 @@ contract WormholeL2ERC20 is ERC20Votes, WormholeReceiver, WormholeSender {
     isRegisteredSender(sourceChain, sourceAddress)
     replayProtect(deliveryHash)
   {
-    (address account, uint256 amount) = abi.decode(payload, (address, uint256));
-    _mint(account, amount);
+    _mint(address(bytes20(payload[:20])), uint224(bytes28(payload[20:48])));
```

Comment:
Should the unlock function below use `encodePacked`?

Reply:
I think the `L2VoteAggregator` might be another candidate for `encodePacked`, and some other spots.

Reply:
> Should the unlock function below use `encodePacked`?

looks like by doing so, it saves ~3000 gas, so sure!

Reply:
> I think the `L2VoteAggregator` might be another candidate for `encodePacked`, and some other spots.

I gave it a shot, but I struggled a bunch with calldata/memory conversion, so I'm wondering if we can defer

Reply:
Yes, let's defer not, also struggled when I played around with it